content/en/docs/hertz/tutorials/advanced-exten/protocol.md

```diff
@@ -0,0 +1,166 @@
+---
+title: "Protocol extension"
+linkTitle: "Protocol extension"
+weight: 2
+description: >
+
+---
+
+## Overview
+
+Because Hertz uses a layered design, in addition to the HTTP1/HTTP2 (to be open source) protocol server that comes with the Hertz framework by default, users can easily add/customize protocol processing logic that meets the needs of their own business scenarios according to their own needs.
+
+A service that implements the following interfaces can be added to Hertz as a custom extension server:
```

Comment:
In short, a server that implements the following interface can be...

server not service